<?xml version="1.0" encoding="UTF-8" standalone="no" ?>
<launch>

	<arg name="ik_file" 	default="/srv/data/gait1992/inverse_kinematics/task_InverseKinematics.mot" />
	<arg name="model_file" 	default="/srv/data/gait1992/residual_reduction_algorithm/model_adjusted.osim" />
	<arg name="rate_divider" default="1" />
	<arg name="async_run"    default="false"/>
	<arg name="simulation_loops"    	default="1"/>


        <node pkg="opensimrt_bridge" type="ik_dump" ns="ik" 	name="inverse_kinematics_from_file" output="screen">
		<param name="ik_file" 			value="$(arg ik_file)" />
                <param name="model_file" 		value="$(arg model_file)" />
                <param name="remove_n_last_rows" 	value="0" />
                <param name="simulation_loops" 		value="$(arg simulation_loops)" />
                <param name="resample_period" 		value="0.01" />
		<!-- maybe you want to run this node slower than the resampling period to give a lot of time between messages for the other algorithms to run. Otherwise this number should be one -->
		<param name="rate_divider" 		value="$(arg rate_divider)" />
		
		<!-- Filter values for filtered output -->
		<param name="filter_output" 		value="true" />
		<rosparam command="load" file="$(find osrt_ros)/config/filter.yaml"/>

		<remap from="/inverse_kinematics_from_file/output" to="/ik/output"/>
		<remap from="/inverse_kinematics_from_file/output_filtered" to="/ik/output_filtered"/>
		<remap from="/inverse_kinematics_from_file/out_labels" to="/ik/out_labels"/>

		<param name="async_run" 			value="$(arg async_run)"/>
        </node>

</launch>
